### ISSUE TITLE
Scoop update fails after install - Windows 7 & 10

### ISSUE THREAD

**Issue by b-palaniappan** [ScoopInstaller/Scoop#2248]
After a successful install in powershell, able to install 7zip and git too.

Then the update fails `scoop update`. I get error,

```
Move-Item : Access to the path 'C:\Users\user2\scoop\apps\scoop\new' is
denied.
At C:\Users\user2\scoop\apps\scoop\current\libexec\scoop-update.ps1:75
char:9
+         Move-Item $newdir $currentdir
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Users\user2\scoop\apps\scoop\
   new:DirectoryInfo) [Move-Item], IOException
    + FullyQualifiedErrorId : MoveDirectoryItemIOError,Microsoft.PowerShell.Co
   mmands.MoveItemCommand

Can't shim 'scoop.ps1': couldn't find 'C:\Users\user2\scoop\apps\scoop\curren
t\bin\scoop.ps1'.
```

when I try to install in Windows 10, it works fine, but fails in Windows 7. I tried in couple of Windows 7 machine. Any help is appreciated.

**Comment by joshlsullivan** [comment]
This is happening to me as well. 

```PS C:\Users\joshlsullivan> scoop update *
Latest versions for all apps are installed! For more information try 'scoop status'
PS C:\Users\joshlsullivan> scoop status
WARN  Scoop is out of date. Run 'scoop update' to get the latest changes.
PS C:\Users\joshlsullivan> scoop update
Updating Scoop...
Move-Item : Access to the path 'C:\Users\joshlsullivan\scoop\apps\scoop\new' is denied.
At C:\Users\joshlsullivan\scoop\apps\scoop\current\libexec\scoop-update.ps1:78 char:9
+         Move-Item $newdir $currentdir
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Users\joshls...\apps\scoop\new:DirectoryInfo) [Move-Item], IOException
    + FullyQualifiedErrorId : MoveDirectoryItemIOError,Microsoft.PowerShell.Commands.MoveItemCommand

Can't shim 'scoop.ps1': couldn't find 'C:\Users\joshlsullivan\scoop\apps\scoop\current\bin\scoop.ps1'.```

**Comment by joshlsullivan** [comment]
Here is it trying to update before installing a package:

`PS C:\Users\joshlsullivan> scoop install postman
Updating Scoop...
Move-Item : Access to the path 'C:\Users\joshlsullivan\scoop\apps\scoop\new' is denied.
At C:\Users\joshlsullivan\scoop\apps\scoop\current\libexec\scoop-update.ps1:78 char:9
+         Move-Item $newdir $currentdir
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Users\joshls...\apps\scoop\new:DirectoryInfo) [Move-Item], IOException
    + FullyQualifiedErrorId : MoveDirectoryItemIOError,Microsoft.PowerShell.Commands.MoveItemCommand

Can't shim 'scoop.ps1': couldn't find 'C:\Users\joshlsullivan\scoop\apps\scoop\current\bin\scoop.ps1'.
You cannot call a method on a null-valued expression.
At C:\Users\joshlsullivan\scoop\apps\scoop\current\lib\depends.ps1:27 char:12
+         if(!(buckets).Contains($bucket)) {
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Couldn't find manifest for 'postman'.
PS C:\Users\joshlsullivan> scoop
& : The term 'C:\Users\joshlsullivan\scoop\shims\..\apps\scoop\current\bin\scoop.ps1' is not recognized as the name of
a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Users\joshlsullivan\scoop\shims\scoop.ps1:2 char:71
+ ... on.expectingInput) { $input | & $path  @args } else { & $path  @args  ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\joshls...t\bin\scoop.ps1:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\joshlsullivan> scoop help
& : The term 'C:\Users\joshlsullivan\scoop\shims\..\apps\scoop\current\bin\scoop.ps1' is not recognized as the name of
a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Users\joshlsullivan\scoop\shims\scoop.ps1:2 char:71
+ ... on.expectingInput) { $input | & $path  @args } else { & $path  @args  ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\joshls...t\bin\scoop.ps1:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException`

**Comment by gregmac** [comment]
It looks like the script `.\scoop\apps\scoop\current\bin\scoop.ps1` is trying to move `.\scoop\apps\scoop\new` to `.\scoop\apps\scoop\current`. After this fails, the `.\scoop\apps\scoop\current` directory is missing (presumably it gets removed) which leaves scoop in a non-working state.

Manually renaming the `new` directory to `current` gets things going again (and scoop is now up-to-date).

**Comment by cprivite** [comment]
Just a note this is still an issue as of a fresh first time install today.

**Comment by RavenMacDaddy** [comment]
I'll post this in the current topics regarding the issue.

---

For me personally it's a number of things:

- The 'new' dir is created but it's empty and Scoop still deletes it and the corresponding bin files etc. in its 'current' dir, making the installation broken.
- There are no permission issues, both regarding what's reported and also confirmed in the properties. When updating Scoop it just halts for a bit and then says "hey, I can't find my files" (it specifically references 'scoop.ps1') on a single line cause it's deleted itself.
- This happens whether or not I'm using forks of the project and regardless of where environment variables are pointing Scoop (where it's installed); could be in its default Users folder dir or somewhere else, doesn't matter, the issue still occurs
- I can reinstall Scoop in the apps folder with the command that was mentioned previously; `iex (new-object net.webclient).downloadstring('https://get.scoop.sh')` - but this only resets the issue, cause Scoop will still try and update when I tell it to and the same issue is repeated
- Doesn't matter if I reinstall Scoop completely any number of times, which I did to try and solve the problem. I tried different combinations mentioned here; installing apps first, looking for .git dir, nothing so far helps.
- The only thing that I managed to do after several tries was to restore my previous installation of Scoop from a backup and eventually after a number of tries and orders of installing apps like 'git' it would behave.
